### PR TITLE
Feature/auto generated metadata

### PR DESCRIPTION
**Pull request by ArielS1**

### CODE REVIEW

File: lib/utils/metadata.js [hunk 1]
```diff
@@ -1,472 +1,792 @@
...
+   const apiMetadata = {};
+
+   Object.keys(allMetadata).forEach(metadataName => {
+       const { minApiVersion, maxApiVersion, xmlName, requiredField, children = {} } = allMetadata[metadataName];
+       if(apiVersion >= minApiVersion && apiVersion <= maxApiVersion) {
+           apiMetadata[metadataName] = { xmlName, requiredField, children };
+       }
+   });
+
+   return apiMetadata;
```
Comment (scolladon) [inline lib/utils/metadata.js]: @ArielS1 I would suggest to use a reduce call in this use case :
```javascript
const apiMetadata = Object.keys(allMetadata).reduce((acc,metadataName) => {
  const { minApiVersion, maxApiVersion, xmlName, requiredField, children = {} } = allMetadata[metadataName];
  if(apiVersion >= minApiVersion && apiVersion <= maxApiVersion) {
    acc[metadataName] = { xmlName, requiredField, children };
  }
 },{});

return apiMetadata;
```

Something like that

Comment (ArielS1) [reply lib/utils/metadata.js]: I tried to do it like this:

`module.exports = apiVersion => {
    apiVersion = parseInt(apiVersion);
    return Object.keys(allMetadata).reduce((acc,metadataName) => {
      const { minApiVersion, maxApiVersion, xmlName, requiredField, children = {} } = allMetadata[metadataName];
      if(apiVersion >= minApiVersion && apiVersion <= maxApiVersion) {
        acc[metadataName] = { xmlName, requiredField, children };
      }
    },{});
}`

But i encountered an error:

`(node:26496) UnhandledPromiseRejectionWarning: TypeError: Cannot set property 'eclair' of undefined
    at Object.keys.reduce (C:\Users\aaa\AppData\Roaming\npm\node_modules\sfdc-generate-package\lib\utils\metadata.js:8:27)
    at Array.reduce (<anonymous>)
    at module.exports.apiVersion (C:\Users\aaa\AppData\Roaming\npm\node_modules\sfdc-generate-package\lib\utils\metadata.js:5:37)
    at Promise (C:\Users\aaa\AppData\Roaming\npm\node_modules\sfdc-generate-package\index.js:20:22)
    at new Promise (<anonymous>)
    at module.exports (C:\Users\aaa\AppData\Roaming\npm\node_modules\sfdc-generate-package\index.js:18:19)
    at Object.<anonymous> (C:\Users\aaa\AppData\Roaming\npm\node_modules\sfdc-generate-package\bin\cli:18:1)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
(node:26496) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 2)
(node:26496) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.`

'eclair' is one of the metadata objects.
i'm not sure why it's happening, but the original method i committed works.

**Comment by scolladon** [comment]
ok,

try this: 
```javascript
const apiMetadata = Object.keys(allMetadata).reduce((acc,metadataName) => {
  const { minApiVersion, maxApiVersion, xmlName, requiredField, children = {} } = allMetadata[metadataName];
  if(apiVersion >= minApiVersion && apiVersion <= maxApiVersion) {
    acc[metadataName] = { xmlName, requiredField, children };
  }
return acc;
 },{});
```

**Comment by ArielS1** [comment]
i implemented reduce as you shown and it works

**Comment by scolladon** [comment]
Thanks for your helpful and elegant pull request @ArielS1 
:)
keep pushing ! 👍

**Comment by scolladon** [comment]
new release :
+ sfdc-generate-package@2.8.0

**Pull request merged**